solution/index.js

```diff
@@ -0,0 +1,278 @@
+
+        switch (id) {
+
+            case 'submit-add-to-do':
+                savedData['todo'].unshift(listValue);
+                break;
+
+            case 'submit-add-in-progress':
+                savedData['in-progress'].unshift(listValue);
+                break;
+
+            case 'submit-add-done':
+                savedData['done'].unshift(listValue);
+                break;
+
+        }
+
+     saveToLocal(savedData);
+}
+
+function searchFilter() // This function works by searching everything with the class of "task" if its includes the text in the searchbar, if it doesnt it removes it from the ul and print all the other ones
+{
+    const search = document.getElementById("search").value.toLowerCase(); 
+    const allTasks= document.querySelectorAll(".task");
+
+    for(li of allTasks)
+    {
+        const liText= li.textContent.toLowerCase() 
+        if(!liText.includes(search))
+        {
+            li.remove();
+        }
+    }
+
+    if(search === "")
+    {
+        for(let li of allTasks)
+        {
+            li.remove();
+        }
+        printData();
+    } 
+}
+
+function editTask (e) { // focuses on the task that you double click, make an input over it, make a new task and delete the changed task. also it adds a CSS class for the last task edited
+    const target = e.target;
+    const mouseEvent = e;
+
+    if (e.target.className === "task" || e.target.className === "last-edited-task")
+    {  
+        const li = target;
+        const oldText = li.textContent;
+        const input = document.createElement("input");
+        input.placeholder = li.textContent;
+        input.classList.add("task");
+        input.focus(); 
+        li.replaceWith(input)
+        input.onblur= (e)=> {
+
+            if (input.value === "")
+            {
+                alert("Please enter text")
+            }
+
+            else 
+            {
+                li.textContent = input.value;
+                input.replaceWith(li); // replaces the list with the new value.
+                const section = li.closest("section");
+                const buttonId = section.querySelector("button").id;
+                const index = keyOfTask(mouseEvent);
+
+                SaveData(buttonId, li.textContent);
+                deleteTask(index, oldText);
+
+            }
+            
+        }
+    }
+}
+
+function moveTask (e) { // makes an object with all the currently pressed keyboard keys and checks if the right keys are pressed the the same time, if they do it takes the same task (duplicate task) put it in the selected section and then removes it from the previus section and saves to local data
+    if (e.target.className === "task" || e.target.className === "last-edited-task")
+    {
+        let keyPressed = {};
+        let mouseEvent = e;
+        document.onkeydown = (key) => 
+        {
+            keyPressed[key.key] = true;
+
+            {
+
+                if(keyPressed["Alt"] && key.key === "1")
```

Comment:
There's a more correct way to check if the alt key is pressed during a keyboard event...